### PR TITLE
Ensure more code is covered by linting 

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
* Replace SCSS Lint with Stylelint (`scss-lint-govuk` is deprecated and superseded by `stylelint-config-gds`)
* Add listing rake task (`bundle exec rake lint` as well as tasks for individual linters aka `bundle exec rake stylelint`)
* Replace `bundle exec rubocop` with `bundle exec rake lint` in staging pipeline so that more code is covered by linting
* Correct syntax in ERB templates
* Correct syntax in SCSS stylesheets

### CODE REVIEW

File: .github/workflows/staging_pipeline.yml [hunk 1]
```diff
@@ -80,9 +80,9 @@ jobs:
           # runs 'bundle install' and caches installed gems automatically
           bundler-cache: true
 
-      - name: Rubocop
+      - name: Lint
         run: |
-          bundle exec rubocop
+          bundle exec rake lint
```
Comment (paulrobertlloyd) [inline .github/workflows/staging_pipeline.yml:97]: Do we want a single linting task in this pipeline, or separate steps for each linter?

Comment (baarkerlounger) [reply .github/workflows/staging_pipeline.yml:97]: Single task is fine I think

**Comment by baarkerlounger** [comment]
@paulrobertlloyd looks like the linter is failing looks like it's all coming from `/vendor` though so we probably just want to exclude that dir from linting.

**Comment by baarkerlounger** [comment]
Stylelint seems to be failing still

**Comment by paulrobertlloyd** [comment]
> @paulrobertlloyd looks like the linter is failing looks like it's all coming from `/vendor` though so we probably just want to exclude that dir from linting.

This one is odd. That exclusion should be inherited from the Rubocop config, but appears to be ignored. Added it explicitly in https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/498/commits/662e5b8e066bc2f5b454a3c52c235b33679e49e9

**Comment by paulrobertlloyd** [comment]
Thank f*** for that. Bloody yarn.

File: .github/workflows/staging_pipeline.yml [hunk 2]
```diff
@@ -80,9 +82,19 @@ jobs:
           # runs 'bundle install' and caches installed gems automatically
           bundler-cache: true
 
-      - name: Rubocop
+      - name: Set up Node.js
```
Comment (baarkerlounger) [inline .github/workflows/staging_pipeline.yml:85]: Need these changes on the production pipeline too I think

Comment (paulrobertlloyd) [reply .github/workflows/staging_pipeline.yml:85]: Have created a separate PR with quality of life improvements for the 2 workflows in #503. Once that’s merged in, will rebase it into this PR and then merge in the specific changes needed to run all linting tasks in the staging workflow.

**Review by baarkerlounger — APPROVED** [review]

**Pull request merged**